security/compliance_operator/compliance-operator-release-notes.adoc

```diff
@@ -0,0 +1,77 @@
++
+* In this release, the `nodeSelector` and `tolerations` in the `ScanSettings` object are now configurable  This enables deployers to configure where the Result Server will run, and thus what node will host the persistent volume that will contain the raw results. This allows us to schedule a pod that makes use of a persistent volume on the master nodes.
++
+* In this release, the Compliance Operator can now remediate `KubeletConfig` objects.
++
+* In this release, an enhancement had been added for content developers. An error is added to a comment for result objects that differentiates between objects that don't exist in the cluster versus objects that can't be fetched.
++
+* In this release, rule objects now contain two new attributes, `checkType` and `description`. These attributes allow the user to see if the rule pertains to a node check or to a platform check and also allows the user to audit what the rule check does.
++
+* In this release, the remediation template now allows multi-value variables. 
++
+* With this update, the Compliance Operator can change remediations based on variables that are set in the compliance profile. This is useful for remediations that include deployment-specific values such as time outs, NTP server host names, or similar. Additionally, the `ComplianceCheckResult` objects now use the label `compliance.openshift.io/check-has-value` that lists the variables a check can use.
++
+* This enhancement removes the requirement to extend an existing Profile in order to create a Tailored Profile. The `extends` field in the `TailoredProfile` CRD is no longer mandatory. You can now select a list of rule objects to create a tailor profile from scratch. Note: You will need to select whether your profile applies to nodes or the platform by setting the `compliance.openshift.io/product-type:` annotation or by setting the `-node` suffix for the `TailoredProfile` CR.
++
+* Previously, only master node taints were tolerated. Now, the Compliance Operator tolerates all taints, making it possible to schedule scans on all nodes.
+
+
+[id="openshift-compliance-operator-0-1-44-bug-fixes"]
+=== Bug fixes
+
+* Previously, while performing a scan, a crash was occurring in one of the scanner containers of the pods performing the scan. In this release, the Compliance Operator uses the latest OpenSCAP version 1.3.5 to avoid this issue.
++
+* Previously, only one global `KubeletConfig` object was created per pool. In this release, a `KubeletConfig` object is created for each pool. 
++
+* The rules for tailored profiles are now validated so that only rules of the same type (Platform or Node) are included in the validation.
++
+* Remediations now have a `NeedsReview` state if they require a variable to be set. In order to automatically apply remediations, you must set all required variables before the MachineConfig pool is unpaused and the remediations applied.
```

Comment:
Ah, sorry, you're right, this was too low level.

So CO has this option called `autoApplyRemediations`, if it's set to true (default is false), then CO will apply all the fixes for all the compliance findings, providing a sort of a one-click compliance. But because of the way the MachineConfigOperator which consumes and actually applies most of the remediations works, applying a remediation triggers an update of the cluster nodes. This is disruptive. To work around that, a MachineConfigPool (a set of machines to apply the remediations to) can be paused, all the needed remediations are marked for application, but really only applied when the pool is unpaused.

Now finally to the point of the note: In case a remediation can't be applied because some required input variable is missing (e.g. the remediation tries to set the NTP servers to be used, which can only be provided by the user), the remediation is assigned a "NeedsReview" state. As long as there are some "NeedsReview" remediations, a pool can't be unpaused.

HTH